### PR TITLE
Signs repository and endpoint

### PR DESCRIPTION
**Pull request by yCobanoglu**

### CODE REVIEW

**Comment by yCobanoglu** [comment]
[Search for signs interface](https://trello.com/c/wDJwbcrk/514-search-for-signs-interface)

File: ebl/tests/transliteration/test_memoizing_sign_repository.py [hunk 1]
```diff
@@ -21,9 +21,78 @@ def test_search_memoization(sign_repository, signs):
     sub_index = sign.values[0].sub_index
 
     memoizing_sign_repository = MemoizingSignRepository(sign_repository)
-    memoizing_sign_repository.create(sign)
+    [memoizing_sign_repository.create(sign) for sign in signs]
```
Comment (jlaasonen) [inline ebl/tests/transliteration/test_memoizing_sign_repository.py:24]: Why use list comprehension when you are not using the list?

Is it necessary to create all the signs to test the memoization?

File: ebl/tests/transliteration/test_sign_repository.py [hunk 2]
```diff
@@ -17,6 +18,41 @@
...
+    sign = Sign(
+        SignName("KUR"),
+        (SignListRecord("ABZ", "03+53"),),
+        (Value("kur", 3), Value("ruk")),
+    )
+    sign_dumped = SignSchema().dump(sign)
+    sign_dumped["name"] = sign_dumped.pop("_id")
+    assert SignSchema(unknown=EXCLUDE).load(data) == sign
+    assert SignSchema().dump(sign) == data
+
```
Comment (jlaasonen) [inline ebl/tests/transliteration/test_sign_repository.py:54]: The schema tests should go to own module.

File: ebl/tests/transliteration/test_sign_repository.py [hunk 3]
```diff
@@ -251,11 +207,84 @@ def test_search(
...
+    sign_igi,
+    mongo_sign_igi,
+    sign_si,
+    mongo_sign_si,
+    sign_si_2,
+    mongo_sign_si_2,
+):
+    database[COLLECTION].insert_many([mongo_sign_igi, mongo_sign_si, mongo_sign_si_2])
+    assert sign_repository.search_all("ši", 1) == [sign_igi]
+    assert sign_repository.search_all("none", 1) == []
```
Comment (jlaasonen) [inline ebl/tests/transliteration/test_sign_repository.py:227]: Add separate test cases for different inputs.

File: ebl/tests/transliteration/test_sign_repository.py [hunk 4]
```diff
@@ -251,11 +207,84 @@ def test_search(
...
+    mongo_sign_igi,
+    sign_si,
+    mongo_sign_si,
+    sign_si_2,
+    mongo_sign_si_2,
+):
+    database[COLLECTION].insert_many([mongo_sign_igi, mongo_sign_si, mongo_sign_si_2])
+
+    assert sign_repository.search_by_id("SI") == [sign_si, sign_si_2]
+    assert sign_repository.search_by_id("none") == []
```
Comment (jlaasonen) [inline ebl/tests/transliteration/test_sign_repository.py:243]: Add separate test cases for different inputs.

File: ebl/tests/transliteration/test_sign_repository.py [hunk 5]
```diff
@@ -251,11 +207,84 @@ def test_search(
...
+    sign_repository,
+    sign_igi,
+    mongo_sign_igi,
+    sign_si,
+    mongo_sign_si,
+    sign_si_2,
+    mongo_sign_si_2,
+):
+    database[COLLECTION].insert_many([mongo_sign_igi])
+    assert sign_repository.search_composite_signs("ši", 1) == [sign_igi]
```
Comment (jlaasonen) [inline ebl/tests/transliteration/test_sign_repository.py:290]: These tests only test the stubs, not the actual implementation.

File: ebl/tests/signs/test_signs_route.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import falcon
+import pytest
+
+from ebl.transliteration.infrastructure.menoizing_sign_repository import (
+    MemoizingSignRepository,
+)
+
+
```
Comment (jlaasonen) [inline ebl/tests/signs/test_signs_route.py:8]: Invalid inputs should also be tested.

File: ebl/transliteration/domain/sign.py [hunk 7]
```diff
@@ -32,8 +32,9 @@ class Sign:
     name: SignName
     lists: Sequence[SignListRecord] = tuple()
     values: Sequence[Value] = tuple()
-    mes_zl: str = ""
+    mes_zl: Optional[str] = None
```
Comment (jlaasonen) [inline ebl/transliteration/domain/sign.py:35]: Is there a difference between `""` and `None`? If not, using `missing` in the schema would remove the need for null checks.

**Review by jlaasonen — CHANGES_REQUESTED** [review]
You created a package for signs. The existing modules for signs should be moved there.

File: ebl/signs/web/sign_search.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    @staticmethod
+    def _parse_params(params):
+        if "subIndex" in params.keys():
+            try:
+                params["subIndex"] = int(params["subIndex"])
+            except ValueError:
+                raise DataError(
+                    f"""subIndex '{params["subIndex"]}' has to be a number"""
+                )
+        return params
```
[the remark this answers is not in the document]
Comment (yCobanoglu) [reply ebl/signs/web/sign_search.py:43]: i made a copy and return that copy now.

**Review by jlaasonen — APPROVED** [review]

**Pull request merged**